app/controllers/topics_controller.rb

```diff
@@ -1314,13 +1313,32 @@ def track_visit_to_topic
       TopicsController.defer_add_incoming_link(hash)
     end
 
-    TopicsController.defer_track_visit(topic_id, ip, user_id, track_visit)
+    TopicsController.defer_track_visit_v2(topic_id, user_id) if should_track_visit_to_topic?
   end
 
+  # TODO (martin) Remove this once discourse-docs is updated.
   def self.defer_track_visit(topic_id, ip, user_id, track_visit)
```

Comment:
So annoying this stuff...I am going to have to do this, then follow these steps in subsequent PRs:

1. Open PR to docs to use the 2 new methods instead of the single `defer_track_visit` (done in https://github.com/discourse/discourse-docs/pull/173)
2. Open another PR to core to change `defer_track_visit` to point to `defer_track_visit_v2`
3. Open another PR to docs to change back to `TopicsController.defer_track_visit(topic_id, user_id)`
4. Make a final core PR to delete `defer_track_visit_v2`